transaction/builder/src/transaction_builder.rs

```diff
@@ -963,34 +962,44 @@ pub mod transaction_builder_tests {
             (BlockVersion::try_from(2).unwrap(), TokenId::from(2)),
         ]
     }
+    // Test memo generator that creates a payload which corresponds to a payment
+    // request.
+    #[derive(Debug)]
+    struct FlexibleMemoGeneratorPaymentRequest {}
+
+    impl FlexibleMemoGeneratorTrait for FlexibleMemoGeneratorPaymentRequest {
```

Comment:
This is the conventional way to implement a trait, and looks good to me. Just rename to `FlexibleMemoGenerator` and add the `Debug` impl discussed above.